### PR TITLE
Update the documentation of the Net module

### PR DESCRIPTION
**Pull request by robertsipka**

### CODE REVIEW

File: docs/api/IoT.js-API-Net.md [hunk 1]
```diff
@@ -43,7 +43,7 @@ However on `nuttx` due to lack of implementation, it does nothing inside.
         - [`'listening'`](#listening)
 - [Class: net.Socket](#class-netsocket)
     - [Constructor](#constructor)
-        - [new net.Socket([options])](#new-netsocketoptions)
+        - [`new net.Socket([options])`](#new-netsocketoptions)
```
Comment (zherczeg) [inline docs/api/IoT.js-API-Net.md]: Please delete the TOC from the file.

File: docs/api/IoT.js-API-Net.md [hunk 2]
```diff
@@ -43,7 +43,7 @@ However on `nuttx` due to lack of implementation, it does nothing inside.
...
     - [Prototype Functions](#prototype-functions)
         - [`socket.connect(options[, connectListener])`](#socketconnectoptions-connectlistener)
         - [`socket.connect(port[, host][, connectListener])`](#socketconnectport-host-connectlistener)
```
Comment (daeyeon) [inline docs/api/IoT.js-API-Net.md]: Remove TOC and its links. `Common  Mistakes` are mentioned in #873.

**Comment by daeyeon** [comment]
- Could you let others know if this is in progress or not through prefixing the title with `[WIP]`?
- If WIP, could you leave brief TODOs like https://github.com/Samsung/iotjs/pull/884?

File: docs/api/IoT.js-API-Net.md [hunk 3]
```diff
@@ -108,61 +273,192 @@ This class is used to create a TCP or local server. You can create `net.Server`
 ## Prototype Funtions
```
Comment (glistening) [inline docs/api/IoT.js-API-Net.md]: @robertsipka Could you please fix the typos (Funtions -> Functions)  although it is not introduced in this PR?

**Comment by glistening** [comment]
@robertsipka Could you check my comment in #873?

File: docs/api/IoT.js-API-Net.md (hunk 1 above)
Comment (robertsipka) [inline docs/api/IoT.js-API-Net.md]: @zherczeg : Thanks, I removed these.

**Comment by robertsipka** [comment]
@daeyeon : It's not a WIP patch.

**Comment by robertsipka** [comment]
@glistening : Thanks, I've updated it accordingly with the last comment on #873 issue, but there are still some suggestions and open question. Leave this PR open until the documents format is not in final state.

**Comment by yichoi** [comment]
LGTM

**Pull request merged**